File: frontend/src/app/model/algorithm-parameters.ts
export class AlgorithmParameters {
  window_size: number;
  threshold: number;
  axis: string;
  peaks_count: number;
  max_readings: number;
  seconds_till_offline: number;
  states_to_swap: number;

  constructor() {
    this.window_size = 0;
    this.threshold = 0;
    this.peaks_count = 0;
    this.max_readings = 0;
    this.seconds_till_offline = 0;
    this.states_to_swap = 0;
    this.axis = 'X';
  }
}

File: backend/src/main/java/com/tablefootbal/server/dsp/AlgorithmParameters.java
package com.tablefootbal.server.dsp;

import com.tablefootbal.server.entity.CalibrationStructure;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.PropertySource;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.stereotype.Component;

import java.io.Serializable;

@Component
@PropertySource("classpath:readings.properties")
@Setter
@Getter
@EqualsAndHashCode
@RedisHash("algorithm_parameters")
public class AlgorithmParameters implements Serializable
{
	@Value("${readings.axis}")
	CalibrationStructure.Axis axis;
	
	@Value("${readings.window_size}")
	private int WINDOW_SIZE;
	
	@Value("${readings.threshold}")
	private double THRESHOLD;
	
	@Value("${readings.minAboveThresholdCount}")
	private int MIN_ABOVE_THRESHOLD_COUNT;
	
	@Value("${readings.max_readings}")
	private int MAX_READINGS;

	@Value("${readings.number_of_states_to_swap}")
	private int NUM_OF_STATES_TO_SWAP;

	@Value("${readings.seconds_until_offline}")
	private int SECONDS_UNTIL_OFFLINE;
}
